code/game/machinery/dna_infuser/dna_infusion.dm

```diff
@@ -0,0 +1,76 @@
+
+///returns a boolean whether a human can be infused
+/atom/movable/proc/can_infuse(mob/feedback_target)
+	if(!isliving(src))
+		if(feedback_target)
+			feedback_target.balloon_alert(feedback_target, "no dna!")
```

Comment:
Also: Should this balloon alerts be sourced on `src`? IE, `balloon_alert(feedback_target, ...)` instead?

Reply:
the reason this is not sourced on src is because infusion targets can be inside objects that might block the visual. for example, the source being inside a DNA infuser's contents.

Reply:
Well I'll be damned!